libs/sdk-ui-dashboard/src/_staging/slideshow/containerSwitcherSlideTransformer.ts

```diff
@@ -0,0 +1,118 @@
+// (C) 2022-2025 GoodData Corporation
+import {
+    IDashboardLayoutItem,
+    IDashboardLayoutSection,
+    isDashboardLayout,
+    isVisualizationSwitcherWidget,
+} from "@gooddata/sdk-model";
+import cloneDeep from "lodash/cloneDeep.js";
+
+import { containsVisualizationSwitcher } from "./utils/index.js";
+import { containerSlideTransformer } from "./containerSlideTransformer.js";
+import { objRef } from "../../model/utils/objRef.js";
+
+/**
+ *
+ * @param item - current item to transform
+ * @param transformLayoutSection - function to transform each section in the layout
+ */
+export function containerSwitcherSlideTransformer<TWidget>(
+    item: IDashboardLayoutItem<TWidget>,
+    transformLayoutSection: (
+        section: IDashboardLayoutSection<TWidget>,
+    ) => IDashboardLayoutSection<TWidget>[] | undefined,
+): IDashboardLayoutSection<TWidget>[] | undefined {
+    if (isDashboardLayout<TWidget>(item.widget)) {
+        // If the layout contains a visualization switcher, iterate over all visualizations in
+        // all switchers and create a slide for each visualization
+        if (containsVisualizationSwitcher(item.widget)) {
+            const totalIterations = countIterations(item.widget, 0);
+            const slides: IDashboardLayoutSection<TWidget>[] = [];
+            for (let i = 0; i < totalIterations; i++) {
+                slides.push({
+                    type: "IDashboardLayoutSection",
+                    items: [switcherIteration<TWidget>(cloneDeep(item), i)],
+                });
+            }
+            return slides;
+        } else {
+            //Default behaviour
+            return containerSlideTransformer(item, transformLayoutSection);
+        }
+    }
+    return undefined;
+}
+
+function countIterations<TWidget>(widget: TWidget, current: number) {
+    if (!isDashboardLayout(widget)) {
+        return current;
+    }
+    for (const section of widget.sections) {
+        for (const item of section.items) {
+            if (isDashboardLayout(item.widget)) {
+                return countIterations(item.widget, current);
+            } else if (isVisualizationSwitcherWidget(item.widget)) {
+                return Math.max(item.widget.visualizations.length, current);
+            }
+        }
+    }
+    return current;
+}
+
+function switcherIteration<TWidget>(
```

Comment:
Again, more descriptive name would be useful (if I understand it correctly, it "flattens" visualization switchers to separate sections?). If so, it can be also likely simplified by using flatMap?

Reply:
No this method basically updated cloned layout and replace all switchers with simple vis on index in switcher. Structure needs to be keep, so flatMap is not usable here